src/Core/Services/ApiService.cs

```diff
@@ -585,6 +585,16 @@ public Task<bool> GetKnownDeviceAsync(string email, string deviceIdentifier)
             });
         }
 
+        public Task<DeviceResponse> GetDeviceByIdentifierAsync(string deviceIdentifier)
+        {
+            return SendAsync<object, DeviceResponse>(HttpMethod.Get, $"/devices/identifier/{deviceIdentifier}", null, true, true);
+        }
+
+        public Task<DeviceResponse> UpdateTrustedDeviceKeysAsync(string deviceIdentifier, TrustedDeviceKeysRequest trustedDeviceKeysRequest)
+        {
+            return SendAsync<TrustedDeviceKeysRequest, DeviceResponse>(HttpMethod.Get, $"/devices/{deviceIdentifier}/keys", trustedDeviceKeysRequest, true, true);
```

Comment:
`HttpMethod.Get` should be `POST` or `PUT`

Reply:
Great catch! Updated